lxqt-admin-user/grpsetmembers.cpp

```diff
@@ -0,0 +1,159 @@
+/* BEGIN_COMMON_COPYRIGHT_HEADER
+ * (c)LGPL2+
+ *
+ * LXQt - a lightweight, Qt based, desktop toolset
+ * http://lxqt.org
+ *
+ * Copyright: 2015 LXQt team
+ * Authors:
+ *   Jesper Schmitz Mouridsen jesper@schmitz.computer
+ *
+ * This program or library is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General
+ * Public License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA
+ *
+ * END_COMMON_COPYRIGHT_HEADER */
+/*
+ * Program setting members specifed by -M of a group specified by -n.
+ * Fails if group or members does not exist.
+ * -M can be empty e.g -M "". It does not add members, it (re)sets the list
+ * of all members. It edits the group file directly.
+ * Written for use with lxqt-admin-user on OpenBSD and NetBSD,
+ * which do not have an easy built in way of maintaining group members.
+  */
+#include <cstdio>
+#include <cstring>
+#include <cstdlib>
+#include <fcntl.h>
+#include <cstdio>
+#include <cstring>
+#include <unistd.h>
+#include <grp.h>
+#include <pwd.h>
+int usage(const char* progname);
+const char * GROUPFILE = "/etc/group";
+const char * GROUPFILE_TMP = "/etc/group.tmp";
+const char * GROUPFILE_BAK = "/etc/group.bak";
+
+int main(int argc, char** argv) {
```

Comment:
Sorry, but I'm not a fan of all this hardcoded manual /etc/group editation... if there is realy no cmd line tool for group manipulation in BSD world, this souldn't be part of the `lxqt-admin`, but a separate project/package